### PR TITLE
filter addon manifestwork with label

### PR DESCRIPTION
**Pull request by zhiweiyin318**
fix issue : https://issues.redhat.com/browse/ACM-4232

### CODE REVIEW

File: pkg/controller/manifestwork/manifestwork_controller.go [hunk 1]
```diff
@@ -169,6 +170,16 @@ func (r *ReconcileManifestWork) deleteManifestWorks(
 	//
 	// if there are any Hosted mode manifestworks we also wait for users to detach the managed cluster first.
 	ignoreKlusterletAndAddons := func(clusterName string, manifestWork workv1.ManifestWork) bool {
+		// if the addon is implemented by addon-framekwork, the format of manifestwork name is
+		// the old format : addon-<addon name>-deploy,
+		// the new format : addon-<addon name>-deploy-<index>
+		// and the manifestwork has the addon label.
+		// so cannot only use the name to filter.
+		workLabels := manifestWork.GetLabels()
+		if _, ok := workLabels[addonapiv1alpha1.AddonLabelKey]; ok {
```
Comment (skeeey) [inline pkg/controller/manifestwork/manifestwork_controller.go:179]: do we need to consider hosted part?

Comment (zhiweiyin318) [reply pkg/controller/manifestwork/manifestwork_controller.go:179]: added

**Comment by skeeey** [comment]
LGTM

/cc @zhujian7

File: pkg/controller/hosted/controller.go [hunk 2]
```diff
@@ -351,6 +352,15 @@ func (r *ReconcileHosted) deleteManifestWorks(ctx context.Context, cluster *clus
 	// delete works that do not include klusterlet works and klusterlet addon works, the addon works were removed
 	// above, we need to wait them to be deleted.
 	ignoreAddons := func(clusterName string, manifestWork workv1.ManifestWork) bool {
+		// if the addon is implemented by addon-framekwork, the format of manifestwork name is
+		// the old format : addon-<addon name>-deploy,
+		// the new format : addon-<addon name>-deploy-<index>
+		// and the manifestwork has the addon label.
+		// so cannot only use the name to filter.
+		workLabels := manifestWork.GetLabels()
+		if _, ok := workLabels[addonapiv1alpha1.AddonLabelKey]; ok {
```
Comment (zhujian7) [inline pkg/controller/hosted/controller.go:361]: for hosted mode addon, I think we need to check the addon namespace label

Comment (zhiweiyin318) [reply pkg/controller/hosted/controller.go:361]: here only delete the manifestworks in hosted cluster ns not hosting cluster ns, so addon name label is ok, right?

**Pull request merged**